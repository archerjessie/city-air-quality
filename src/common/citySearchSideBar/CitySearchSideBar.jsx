import React from 'react';
import SimpleForm from '../simpleForm/SimpleForm';
import { fetchCity } from '../../api/airQualityAPI';
import * as R from 'ramda';
import withState from '../../utils/withState';
import CityLocationButton from './cityLocationButton/CityLocationButton';

import './citySearchSideBar.css';

const CitySearchSideBar = ({
    onLocationSelected,
    onRefreshData,
    searchData,
    setSearchData }) => {
    const onSearchCity = (search) => {
        onRefreshData();
        fetchCity(search, setSearchData);
    }

    return (
        <div className='city-search-sidebar'>
            <SimpleForm onSubmit={onSearchCity} />
            <div className="br1 br--bottom flex flex-column w5 bb bl br b--light-gray border-box">
                {searchData ? (
                    <div className="pv2">
                        {searchData.map(({ uid, station: { name } }) => (
                            <CityLocationButton key={uid} onClick={onLocationSelected} name={name} uid={uid} />
                        ))}
                    </div>
                ) : (
                        <div className="ph3 pv2 silver h4 flex items-center justify-center">
                            Search for a city
                        </div>
                    )}
            </div>
        </div>
    )
}
export default R.compose(
    withState('searchData', 'setSearchData', null),
)(CitySearchSideBar);