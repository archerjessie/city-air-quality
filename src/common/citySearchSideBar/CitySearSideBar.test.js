import React from 'react';
import CitySearchSideBar from './CitySearchSideBar';
import renderer from 'react-test-renderer';

describe("CitySearchSideBar", () => {
    it("renders as expected", () => {
        const searchData = [{ "uid": 3234, "aqi": "16", "time": { "tz": "+10:00", "stime": "2020-10-08 17:00:00", "vtime": 1602140400 }, "station": { "name": "Alphington", "geo": [-37.7784081, 145.0306], "url": "australia/melbourne/alphington", "country": "AU" } }, { "uid": 10141, "aqi": "11", "time": { "tz": "+10:00", "stime": "2020-10-08 17:00:00", "vtime": 1602140400 }, "station": { "name": "Melbourne CBD", "geo": [-37.8073959, 144.97], "url": "australia/melbourne/melbourne-cbd", "country": "AU" } }, { "uid": 3245, "aqi": "-", "time": { "tz": "+10:00", "stime": "2020-02-28 16:00:00", "vtime": 1582869600 }, "station": { "name": "Yarraville", "geo": [-37.816608, 144.890106], "url": "australia/melbourne/yarraville", "country": "AU" } }, { "uid": 3246, "aqi": "-", "time": { "tz": "+10:00", "stime": "2020-02-28 15:00:00", "vtime": 1582866000 }, "station": { "name": "Richmond", "geo": [-37.818635, 145.0014704], "url": "australia/melbourne/richmond", "country": "AU" } }, { "uid": 6283, "aqi": "37", "time": { "tz": "-06:00", "stime": "2020-10-08 01:00:00", "vtime": 1602140400 }, "station": { "name": "Melbourne, Brevard, Florida, USA", "geo": [28.053611111111, -80.628611111111], "url": "usa/florida/brevard/melbourne", "country": "US" } }, { "uid": 3242, "aqi": "17", "time": { "tz": "+10:00", "stime": "2020-10-08 17:00:00", "vtime": 1602140400 }, "station": { "name": "Footscray", "geo": [-37.80488, 144.8728], "url": "australia/melbourne/footscray", "country": "AU" } }, { "uid": 3236, "aqi": "1", "time": { "tz": "+10:00", "stime": "2020-10-08 06:00:00", "vtime": 1602100800 }, "station": { "name": "Brighton", "geo": [-37.9135475, 144.998], "url": "australia/melbourne/brighton", "country": "AU" } }, { "uid": 3239, "aqi": "-", "time": { "tz": "+10:00", "stime": "2020-02-28 16:00:00", "vtime": 1582869600 }, "station": { "name": "AltonaNorth", "geo": [-37.84455, 144.8629], "url": "australia/melbourne/altonanorth", "country": "AU" } }]
        const tree = renderer
            .create(<CitySearchSideBar searchData={searchData} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    })
})