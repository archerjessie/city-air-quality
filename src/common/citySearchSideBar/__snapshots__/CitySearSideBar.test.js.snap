// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitySearchSideBar renders as expected 1`] = `
<div
  className="city-search-sidebar"
>
  <form
    className="flex"
    onSubmit={[Function]}
  >
    <div
      className="w5 flex items-center"
    >
      <input
        className="br1 br--top br--left ba b--light-gray ph3 pv2 flex-grow-1 flex-shrink-1"
        onChange={[Function]}
        placeholder="Melbourne"
        type="text"
        value=""
      />
      <button
        className="br1 bl-0 br--top br--right b--light-gray white pa2 right-0 flex-grow-0 flex-shrink-0"
        type="submit"
      >
        <span
          aria-label="search"
          role="img"
        >
          üïµÔ∏è
        </span>
      </button>
    </div>
  </form>
  <div
    className="br1 br--bottom flex flex-column w5 bb bl br b--light-gray border-box"
  >
    <div
      className="ph3 pv2 silver h4 flex items-center justify-center"
    >
      Search for a city
    </div>
  </div>
</div>
`;
